src/java/build.gradle

```diff
@@ -0,0 +1,249 @@
+
+// cmake runs will inform us of the build directory of the current run
+def cmakeBuildDir = System.properties['cmakeBuildDir']
+def useCUDA = System.properties['USE_CUDA']
+def cmakeJavaDir = "${cmakeBuildDir}"
+def cmakeNativeLibDir = System.properties['nativeLibDir']
+def cmakeBuildOutputDir = "${cmakeJavaDir}/build"
+
+def mavenUser = System.properties['mavenUser']
+def mavenPwd = System.properties['mavenPwd']
+def mavenArtifactId = useCUDA == null ? project.name : project.name + "_gpu"
+
+def defaultDescription = 'ONNX Runtime GenAI is <TODO>'
+
+logger.lifecycle("cmakeBuildDir:${cmakeBuildDir}")
+logger.lifecycle("cmakeNativeLibDir:${cmakeNativeLibDir}")
+
+java {
+	sourceCompatibility = JavaVersion.VERSION_1_8
+	targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+// This jar tasks serves as a CMAKE signalling
+// mechanism. The jar will be overwritten by allJar task
+jar {
+}
+
+// Add explicit sources jar with pom file.
+task sourcesJar(type: Jar, dependsOn: classes) {
+	archiveClassifier = "sources"
+	from sourceSets.main.allSource
+	into("META-INF/maven/$project.group/$mavenArtifactId") {
+		from { generatePomFileForMavenPublication }
+		rename ".*", "pom.xml"
+	}
+}
+
+// Add explicit javadoc jar with pom file
+task javadocJar(type: Jar, dependsOn: javadoc) {
+	archiveClassifier = "javadoc"
+	from javadoc.destinationDir
+	into("META-INF/maven/$project.group/$mavenArtifactId") {
+		from { generatePomFileForMavenPublication }
+		rename ".*", "pom.xml"
+	}
+}
+
+spotless {
+	java {
+		removeUnusedImports()
+		googleJavaFormat()
+	}
+	format 'gradle', {
+		target '**/*.gradle'
+		trimTrailingWhitespace()
+		indentWithTabs()
+
+	}
+}
+
+compileJava {
+	dependsOn spotlessJava
+	options.compilerArgs += ["-h", "${project.buildDir}/headers/"]
+	if (!JavaVersion.current().isJava8()) {
+		// Ensures only methods present in Java 8 are used
+		options.compilerArgs.addAll(['--release', '8'])
+		// Gradle versions before 6.6 require that these flags are unset when using "-release"
+		java.sourceCompatibility = null
+		java.targetCompatibility = null
+	}
+}
+
+compileTestJava {
+	if (!JavaVersion.current().isJava8()) {
+		// Ensures only methods present in Java 8 are used
+		options.compilerArgs.addAll(['--release', '8'])
+		// Gradle versions before 6.6 require that these flags are unset when using "-release"
+		java.sourceCompatibility = null
+		java.targetCompatibility = null
+	}
+}
+
+sourceSets.main.java {
+	srcDirs = ['src/main/java', 'src/main/jvm']
+}
+
+// expects
+sourceSets.test {
+	// add test resource files
+	resources.srcDirs += [
+		"../../test/test_models"
+	]
+
+	if (cmakeNativeLibDir != null) {
+		// add compiled native libs
+		resources.srcDirs += [cmakeNativeLibDir]
+	}
+}
+
+if (cmakeNativeLibDir != null) {
+	// generate tasks to be called from cmake
+
+	// Overwrite jar location
+	task allJar(type: Jar) {
+		manifest {
+			attributes('Automatic-Module-Name': project.group,
```

Comment:
I'd missed this initially & checked again after noticing a similar issue in ORT-extensions (https://github.com/microsoft/onnxruntime-extensions/pull/730), but `com.microsoft.onnxruntime-genai` isn't a valid Java module name as it has a hypen in. Could this be changed to `com.microsoft.onnxruntime.genai`?